### DISCUSSION TITLE
API RESTFULL [ ABSTRATA X CONCRETA ]

### DISCUSSION THREAD

**Discussion by abalad** [frontendbr/forum#468] [Discussão]
Ao Desenvolver uma API RestFull o que seria melhor:

- **Uma API Abstrata(Generica)** onde cada recurso é generico de acordo com as tabelas/coleções de um Banco de Dados, utilizando a mesma base de Código sem precisar escrever um método para cada recurso, alterando apenas a configuração de banco. _Exemplo https://www.dreamfactory.com/_
 
- **Uma API Concreta** onde é criado um método para cada recurso de um determinado banco em especifico, tornando aquela API unica para ele projeto.

**Comment by Lucasktrindade** [comment]
Depende do projeto, na minha visão poderia usar a abstrata para fazer integração entre sistemas, por exemplo você tem um serviço e oferece para uma empresa que já tem um sistema completo. Agora quando você ta desenvolvendo a solução completa eu ia preferir fazer uma API Concreta.
